rust_icu_uloc/src/lib.rs

```diff
@@ -239,13 +295,22 @@ impl ULoc {
         ffi::CString::new(self.repr.clone()).expect("ULoc contained interior NUL bytes")
     }
 
-    // Implements `uloc_acceptLanguage` from ICU4C.
-    #[deprecated = "Use `rust_icu_uloc::accept_language`"]
-    pub fn accept_language(
-        accept_list: impl IntoIterator<Item = impl Into<ULoc>>,
-        available_locales: impl IntoIterator<Item = impl Into<ULoc>>,
-    ) -> Result<(Option<ULoc>, UAcceptResult), common::Error> {
-        accept_language(accept_list, available_locales)
+    /// Implements `uloc_forLanguageTag` from ICU4C.
+    pub fn for_language_tag(tag: &str) -> Result<ULoc, common::Error> {
+        buffered_string_method_with_retry!(
+            buffered_string_for_language_tag,
+            LOCALE_CAPACITY,
+            [tag: *const raw::c_char,],
+            [parsed_length: *mut i32,]
+        );
+
+        let tag = ffi::CString::new(tag).map_err(|_| common::Error::string_with_interior_nul())?;
```

Comment:
For example, this is one place where it's OK to leave the error conversion in place, since the constructor is allowed to produce an error.

But... it reminds me... should we have an automated `.into()` for `NulError` ? It seems possible, and the reason I didn't do this initially was mostly PEBKAC, not a real need.

Reply:
Added `From<NulError>` and `From<Utf8Error>` implementations for `common::Error`.